src/Extension/CoreExtension.php

```diff
@@ -895,6 +896,36 @@ public static function reverse(string $charset, $item, $preserveKeys = false)
         return $string;
     }
 
+    /**
+     * Shuffles an array, a \Traversable instance, or a string.
+     * The function does not preserve keys.
+     *
+     * @internal
+     */
+    public static function shuffle(string $charset, array|\Traversable|string|null $item): mixed
```

Comment:
As Twig 3.x supports older versions of PHP, the `array|\Traversable|string|null` type hint should be a phpdoc instead.